utils.js

```diff
@@ -117,49 +137,63 @@ const getUserTotalVotingPowerAtReferenceBlock = async (
   daoContract,
   tokenID,
   level,
-  userAddress
+  userAddress,
+  isXTZ
 ) => {
   try {
-    let userVotingPower = new BigNumber(0);
 
-    const isTokenDelegation = await isTokenDelegationSupported(
-      network,
-      address
-    );
+    if (!isXTZ) {
+      let userVotingPower = new BigNumber(0);
 
-    if (isTokenDelegation) {
-      const userVotePower = await getUserTotalVotingWeightAtBlock(
+      const isTokenDelegation = await isTokenDelegationSupported(
         network,
-        address,
-        level,
-        userAddress
+        address
       );
-      if (!userVotePower) {
-        throw new Error("Could Not get voting weight");
+
+      if (isTokenDelegation) {
+        const userVotePower = await getUserTotalVotingWeightAtBlock(
+          network,
+          address,
+          level,
+          userAddress
+        );
+        if (!userVotePower) {
+          throw new Error("Could Not get voting weight");
+        }
+        userVotingPower = userVotingPower.plus(userVotePower);
+      } else {
+        const selfBalance = await getUserBalanceAtLevel(
+          network,
+          address,
+          tokenID,
+          level,
+          userAddress
+        );
+        userVotingPower = userVotingPower.plus(selfBalance);
+
+        if (daoContract) {
+          const userDAODepositBalance = await getUserDAODepositBalanceAtLevel(
+            userAddress,
+            network,
+            daoContract,
+            level
+          );
+          userVotingPower = userVotingPower.plus(userDAODepositBalance);
+        }
       }
-      userVotingPower = userVotingPower.plus(userVotePower);
+
+      return userVotingPower;
```

Comment:
return userVotingPower - can be combined to be outside if

Reply:
fixed!